### PR TITLE
V68 implementation: refunds + reversals

### PR DESCRIPTION
**Pull request by vadim-nilov**
**Description**
Added path parameters handler that allows implementing the rest methods of the Checkout API integration.

Added `v68` Checkout methods implementation:
- POST [/payments/{paymentPspReference}/refunds](https://docs.adyen.com/api-explorer/#/CheckoutService/v68/post/payments/{paymentPspReference}/refunds)
- POST [/payments/{paymentPspReference}/reversals](https://docs.adyen.com/api-explorer/#/CheckoutService/v68/post/payments/{paymentPspReference}/reversals)

**Tested scenarios**
Both methods have integration test:
- \Adyen\Tests\Integration\CheckoutTest::testRefunds
- \Adyen\Tests\Integration\CheckoutTest::testReversals

### CODE REVIEW

**Comment by vadim-nilov** [comment]
Hi there, is there anybody who can check the PR? Could you let me know if you don't plan to do that, thanks

**Comment by peterojo** [comment]
Hi @vadim-nilov apologies for the delay, we have been working on some changes in #331 and wanted to make sure we merge that before considering further changes to the resource model.
Please look into it and resolve the conflicts then we will review this PR.
Thanks again for the contribution :)

**Comment by vadim-nilov** [comment]
> Hi @vadim-nilov apologies for the delay, we have been working on some changes in #331 and wanted to make sure we merge that before considering further changes to the resource model. Please look into it and resolve the conflicts then we will review this PR. Thanks again for the contribution :)

Thanks for the response! Conflicts were resolved

File: src/Adyen/Service/ResourceModel/Checkout/Refunds.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Payments constructor.
+     *
+     * @param \Adyen\Service $service
+     * @throws \Adyen\AdyenException
+     */
+    public function __construct($service)
+    {
+        $this->endpoint = $this->getCheckoutEndpoint($service) .
+            '/' . $service->getClient()->getApiCheckoutVersion() . '/payments/{paymentPspReference}/refunds';
```
Comment (AlexandrosMor) [inline src/Adyen/Service/ResourceModel/Checkout/Refunds.php:28]: How do you pass the paymentPspReference in the url ?

File: src/Adyen/Service/ResourceModel/Checkout/Reversals.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Payments constructor.
+     *
+     * @param \Adyen\Service $service
+     * @throws \Adyen\AdyenException
+     */
+    public function __construct($service)
+    {
+        $this->endpoint = $this->getCheckoutEndpoint($service) .
+            '/' . $service->getClient()->getApiCheckoutVersion() . '/payments/{paymentPspReference}/reversals';
```
Comment (AlexandrosMor) [inline src/Adyen/Service/ResourceModel/Checkout/Reversals.php:28]: Same here, How do you pass the `paymentPspReference` in the url ?

File: src/Adyen/Service/ResourceModel/Checkout/Refunds.php (hunk 1 above)
Comment (vadim-nilov) [reply src/Adyen/Service/ResourceModel/Checkout/Refunds.php:28]: Hi @AlexandrosMor, I pass it as a request param, see: `\Adyen\Service\AbstractResource::replacePathParameters`

If there is a match then I replace it.

You can check the example in the integration test:
`\Adyen\Tests\Integration\CheckoutTest::testRefunds`

**Review by peterojo — APPROVED** [review]

**Review by AlexandrosMor — DISMISSED** [review]

**Comment by vadim-nilov** [comment]
Hi there, I fixed one typo, now it should pass code sniffer checks

**Review by peterojo — APPROVED** [review]

**Comment by vadim-nilov** [comment]
@acampos1916, @rikterbeek, @michaelpaul, @AlexandrosMor, @Morerice could you review my PR when you have time? Thanks!

**Review by acampos1916 — APPROVED** [review]

**Pull request merged**